### ISSUE TITLE
Ability to enable/disable the plugins without changing the config

### ISSUE THREAD

**Issue by fredwangwang** [influxdata/telegraf#10543]
## Feature Request

Ability to enable/disable the plugin without changing the config (using env vars)

### Proposal:

add a common `enabled` flag to all plugins.

```
# Read metrics about memory usage
[[inputs.mem]]
enabled = ${MEM_INPUTS_ENABLED}

# Get the number of processes and group them by status
[[inputs.processes]]
enabled = ${PROCESSES_INPUTS_ENABLED}
```

### Current behavior:

not supported

### Desired behavior:

the above example config can selectively enable inputs based on envvars without making changes to the config, make it a bit more 12-factor-ly

### Use case: 

Right now each plugin can be enabled/disabled explicitly by adding/removing sections in the configuration file, however for our use case we want to control the enablement of the plugin through env vars, like we have for [filebeat](https://www.elastic.co/guide/en/beats/filebeat/6.8/file-output.html#_enabled_13)

Because of telegraf doesn't support this ability right now, we use a pre-start script to template the config before starting the telegraf instance, it would be great to not have to do that :)

**Comment by srebhan** [comment]
Hey @fredwangwang, I'm using a slightly different approach. I'm using the `--config-dir` option with config files that only contain a certain group of plugins (all mandatory inputs, different groups of optional inputs, processors, production outputs, debug outputs, staging outputs,...). Then I'm symlinking these config to different configurations (in different subdirs) and start telegraf with the corresponding set of subdirs... Would that also work for you?

**Comment by fredwangwang** [comment]
Hi @srebhan thanks for ur reply! Yes the approach you described definitely works, but it also requires some pre-start manipulation of the files etc, which is what we have as well :D

What I am looking for is getting rid of the pre-start script altogether. It is certainly not affecting any functionalities we are trying to achieve without this feature, but purely as a nice to have because filebeat supports this kinda enablement, that we want to adapt the same style for telegraf as for filebeat.

**Comment by srebhan** [comment]
@fredwangwang understood. An alternative to what you want is to use the `--input-filter`, `--output-filter`, ... [options of telegraf](https://github.com/influxdata/telegraf/blob/master/docs/COMMANDS_AND_FLAGS.md#flags). This could be passed in by environment variables I think...

**Comment by fredwangwang** [comment]
oh nice! I thought `--output-filter` is only used when generating the configuration, this is actually a nice way for my use case, thanks!

@srebhan Going to leave it up to you to keep this FR or close it then

**Comment by srebhan** [comment]
As we are currently not running out of FRs and there is a viable solution already I'm closing this. :-)
